cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -0,0 +1,276 @@
+		return nil, err
+	}
+	gitRepoDetails.gitProjectPath = i.azdCtx.ProjectDirectory()
+	return gitRepoDetails, nil
+}
+
+func (i *pipelineManager) getGitRepoDetails(ctx context.Context) (*gitRepositoryDetails, error) {
+	gitCli := git.NewGitCli()
+	repoPath := i.azdCtx.ProjectDirectory()
+	for {
+		repoRemoteDetails, err := i.ensureRemote(ctx, repoPath, i.pipelineRemoteName)
+		switch {
+		case errors.Is(err, git.ErrNotRepository):
+			// Offer the user a chance to init a new repository if one does not exist.
+			initRepo, err := i.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      "Initialize a new git repository?",
+				DefaultValue: true,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("prompting for git init: %w", err)
+			}
+
+			if !initRepo {
+				return nil, errors.New("confirmation declined")
+			}
+
+			if err := gitCli.InitRepo(ctx, repoPath); err != nil {
+				return nil, fmt.Errorf("initializing repository: %w", err)
+			}
+
+			// Recovered from this error, try again
+			continue
+		case errors.Is(err, git.ErrNoSuchRemote):
+			// Offer the user a chance to create the remote if one does not exist.
+			addRemote, err := i.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      fmt.Sprintf("A remote named \"%s\" was not found. Would you like to configure one?", i.pipelineRemoteName),
+				DefaultValue: true,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("prompting for remote init: %w", err)
+			}
+
+			if !addRemote {
+				return nil, errors.New("confirmation declined")
+			}
+
+			// the scm provider returns the repo url that is used as git remote
+			remoteUrl, err := i.scmProvider.configureGitRemote(ctx, repoPath, i.pipelineRemoteName, i.console)
+			if err != nil {
+				return nil, err
+			}
+
+			// set the git remote for local git project
+			if err := gitCli.AddRemote(ctx, repoPath, i.pipelineRemoteName, remoteUrl); err != nil {
+				return nil, fmt.Errorf("initializing repository: %w", err)
+			}
+
+			continue
+		case err != nil:
+			return nil, err
+		default:
+			return repoRemoteDetails, nil
+		}
+	}
+}
+
+// validateDependencyInjection panic if the manager did not received all the
+// mandatory dependencies to work
+func validateDependencyInjection(manager *pipelineManager) {
+	if manager.azdCtx == nil {
+		log.Panic("missing azd context for pipeline manager")
+	}
+	if manager.scmProvider == nil {
+		log.Panic("missing scm provider for pipeline manager")
+	}
+	if manager.ciProvider == nil {
+		log.Panic("missing CI provider for pipeline manager")
+	}
+}
+
+func (i *pipelineManager) pushGitRepo(ctx context.Context, currentBranch string) error {
+	gitCli := git.NewGitCli()
+
+	if err := gitCli.AddFile(ctx, i.azdCtx.ProjectDirectory(), "."); err != nil {
+		return fmt.Errorf("adding files: %w", err)
+	}
+
+	if err := gitCli.Commit(ctx, i.azdCtx.ProjectDirectory(), "Configure GitHub Actions"); err != nil {
+		return fmt.Errorf("commit changes: %w", err)
+	}
+
+	fmt.Println("Pushing changes")
```

Comment:
Should we be calling something on the `console` object instead of writing directly to stdout?

Reply:
Sure.
This code is coming from the current implementation.  I can update it, for sure